src/test/java/core/basesyntax/FruitShopTest.java

```diff
@@ -0,0 +1,189 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import core.basesyntax.service.FileReader;
+import core.basesyntax.service.FileWriter;
+import core.basesyntax.service.ReportGenerator;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.serviceimpl.DataConverterImpl;
+import core.basesyntax.serviceimpl.FileReaderImpl;
+import core.basesyntax.serviceimpl.FileWriterImpl;
+import core.basesyntax.serviceimpl.OperationStrategyImpl;
+import core.basesyntax.serviceimpl.ReportGeneratorImpl;
+import core.basesyntax.serviceimpl.ShopServiceImpl;
+import core.basesyntax.strategy.BalanceOperation;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.PurchaseOperation;
+import core.basesyntax.strategy.ReturnOperation;
+import core.basesyntax.strategy.SupplyOperation;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class FruitShopTest {
+    private FileReader fileReader;
+    private FileWriter fileWriter;
+    private DataConverter dataConverter;
+    private ShopService shopService;
+    private ReportGenerator reportGenerator;
```

Comment:
You should test each service in a separate test class. It should have the same structure as the main solution